sympy/assumptions/ask.py

```diff
@@ -129,6 +131,9 @@ def ask(proposition, assumptions=True, context=global_assumptions):
     else:
         key, expr = Q.is_true, sympify(proposition)
 
+    if isinstance(proposition, (bool, BooleanAtom)):
+        return sympify(proposition)
```

Comment:
I don't remember why I made this change. I agree that ask should return a bool, since it doesn't make sense in a boolean context (https://github.com/sympy/sympy/blob/98459c58b4a99ca01d81a6c485c1778f621eac0e/sympy/logic/boolalg.py#L90-L93). The commit this comes from doesn't specify why it was done. I'm assuming I put it in there because something was not working without it. I would try to remove it and if the tests pass then it can be removed.